aws/hll/aws-s3-transfer-manager/src/download.rs

```diff
@@ -34,17 +49,178 @@ impl From<GetObjectInputBuilder> for DownloadRequest {
     }
 }
 
-/// Response type for a single download object request.
-#[derive(Debug)]
-#[non_exhaustive]
-pub struct DownloadResponse {
-    /// Object metadata
-    pub object_meta: ObjectMetadata,
+/// Fluent style builder for [Downloader]
+#[derive(Debug, Clone)]
+pub struct Builder {
+    target_part_size_bytes: u64,
+    checksum_validation_enabled: bool,
+    // TODO(design): should we instead consider an enum here allows for not only explicit but also
+    // an "Auto" mode that allows us to control the concurrency actually used based on overall transfer and part size?
+    concurrency: usize,
+    sdk_config: Option<SdkConfig>,
+}
+
+impl Builder {
+    fn new() -> Self {
+        Self {
+            target_part_size_bytes: 8 * MEBIBYTE,
+            checksum_validation_enabled: true,
+            concurrency: 8,
+            sdk_config: None,
+        }
+    }
+
+    /// Size of parts the object will be downloaded in, in bytes.
+    /// The minimum part size is 5 MiB and any value given less than that will be rounded up.
+    /// Defaults is 8 MiB.
+    pub fn target_part_size(mut self, size_bytes: u64) -> Self {
+        self.target_part_size_bytes = cmp::min(size_bytes, MIN_PART_SIZE);
+        self
+    }
+
+    /// Set whether checksum validation is enabled.
+    /// Default is true.
+    pub fn enable_checksum_validation(mut self, enabled: bool) -> Self {
+        self.checksum_validation_enabled = enabled;
+        self
+    }
+
+    /// Set the configuration used by the S3 client
+    pub fn sdk_config(mut self, config: SdkConfig) -> Self {
+        self.sdk_config = Some(config);
+        self
+    }
+
+    /// Set the concurrency level this component is allowed to use. This sets the maximum
+    /// number of concurrent in-flight requests.
+    /// Default is 8.
+    pub fn concurrency(mut self, concurrency: usize) -> Self {
+        self.concurrency = concurrency;
+        self
+    }
```

Comment:
TODO Create a breadcrumb leading users to a guide explaining how to choose the right amount of concurrency.

Reply:
Possibly worth explaining the difference between concurrency and parallelism somewhere.

Reply:
I don't know that we know this yet but agreed we'll want to revisit all docs anyway (I am trying to document as we go though to the extent that makes sense).